errors/multi.go

```diff
@@ -96,3 +97,29 @@ func (e multiErr) Contains(err *Error) bool {
 	}
 	return false
 }
+
+// abciLogFrame defines the frame type
+type abciLogFrame struct {
+	Data []abciLogElement `json:"data"`
+}
+
+// abciLogElement represents an error element of the multiErr
+type abciLogElement struct {
+	Code uint32 `json:"code"`
+	Log  string `json:"log"`
+}
+
+// serializeMultiErr converts the given error into a json structured byte string
+func serializeMultiErr(source multiErr, enc errEncoder) string {
+	logs := make([]abciLogElement, len(source))
+	for i, err := range source {
+		code := abciCode(err)
+		logs[i] = abciLogElement{Code: code, Log: enc(err)}
```

Comment:
Very nice... so each error is encoded via debug or default based on the context passed into multiErr.

I like this design a lot. Handle many cases orthogonally.